leveldb/session.go

```diff
@@ -226,9 +230,16 @@ func (s *session) commit(r *sessionRecord, trivial bool) (err error) {
 	if s.manifest == nil {
 		// manifest journal writer not yet created, create one
 		err = s.newManifest(r, nv)
-	} else if s.manifest.Size() >= s.o.GetMaxManifestFileSize() {
+	} else if s.manifest.Size() >= s.maxManifestFileSize {
 		// pass nil sessionRecord to avoid over-reference table file
 		err = s.newManifest(nil, nv)
+		if err == nil {
+			// increase the limit if the result manifest still over-sized, to prevent creation
+			// on each commit.
+			for s.manifest.Size() >= s.maxManifestFileSize {
+				s.maxManifestFileSize *= 2
+			}
```

Comment:
```suggestion
			if manifestSize := s.manifest.Size(); manifestSize >= s.maxManifestFileSize {
				s.maxManifestFileSize = 2 * manifestSize
			}
```

Reply:
I think for loop is need here for the case that manifestSize is much larger than the limit.

Reply:
good suggestion